### ISSUE TITLE
DateTime @db.Date: Udpate issue

### ISSUE THREAD

**Issue by elvince** [prisma/prisma#25455]
### Bug description

hi,
with a postgresql, it is not working as expected.
error: Invalid value for argument `dateOfBirth`: premature end of input. Expected ISO-8601 DateTime
during an update when using:  dateOfBirth: "2024-03-30"

### How to reproduce

using this model : 
model test{
dateOfBirth DateTime? @db.Date
}
 call an update on test with dateOfBirth: "2024-03-30".

### Expected behavior

No error.
Date should not expecttime at all.

### Prisma information

model test{
dateOfBirth DateTime? @db.Date
}

### Environment & setup

- OS: Windows,
- Database: PostgreSQL
- Node.js version: v20.9.0

### Prisma Version

```
prisma                  : 5.20.0
@prisma/client          : 5.20.0
Computed binaryTarget   : windows
Operating System        : win32
Architecture            : x64
Node.js                 : v20.9.0
Query Engine (Node-API) : libquery-engine 06fc58a368dc7be9fbbbe894adf8d445d208c284 (at node_modules\@prisma\engines\query_engine-windows.dll.node)
Schema Engine           : schema-engine-cli 06fc58a368dc7be9fbbbe894adf8d445d208c284 (at node_modules\@prisma\engines\schema-engine-windows.exe)
Schema Wasm             : @prisma/prisma-schema-wasm 5.20.0-12.06fc58a368dc7be9fbbbe894adf8d445d208c284
Default Engines Hash    : 06fc58a368dc7be9fbbbe894adf8d445d208c284
Studio                  : 0.502.0
Preview Features        : views
```

**Comment by zackdotcomputer** [comment]
Agreed that this shouldn't be the behavior. It would also be great if we had the option to define this columns as getting pulled from the DB into JS as type `string` instead of `Date` to avoid any possibility of confusion re times and timezones.

**Comment by oceandrama** [comment]
Can confirm the same behavior and would like to draw attention to this issue

**Comment by wilau2** [comment]
Getting this issue trying to use `@prisma/adapter-pg` 6.10.1
